src/layers/classifier.jl

```diff
@@ -0,0 +1,25 @@
+"""
+    create_classifier(inplanes, nclasses; pool_type = :mean, use_conv = false)
+
+Creates a classifier head to be used for models. Uses `SelectAdaptivePool` for the pooling layer.
+
+# Arguments
+
+  - `inplanes`: number of input feature maps
+  - `nclasses`: number of output classes
+  - `pool_type`: the type of adaptive pooling to use. One of `:mean`, `:max`, `:meanmax`,
+    `:catmeanmax` or `:identity`.
+  - `use_conv`: whether to use a 1x1 convolutional layer instead of a `Dense` layer.
+"""
+function create_classifier(inplanes, nclasses; pool_type = :mean, use_conv = false)
+    flatten_in_pool = !use_conv  # flatten when we use a Dense layer after pooling
+    if pool_type == :identity
+        @assert use_conv
+        "Pooling can only be disabled if classifier is also removed or a convolution-based classifier is used"
+        flatten_in_pool = false  # disable flattening if pooling is pass-through (no pooling)
+    end
+    global_pool = SelectAdaptivePool(; pool_type, flatten = flatten_in_pool)
```

Comment:
This makes sense. Times like this I wish Flux had an abstract type for layers (and for stuff like pooling layers). Then I could define custom pooling types in Metalhead subtyping those and ensure that users didn't pass in random pooling functions to the classifier. One of the primary reasons of introducing `SelectAdaptivePool` was to ensure this - that the case of no pooling - `identity` was handled along with the pooling layer choices

Reply:
I've refactored this to make it so that the user has to pass in the entire layer. Saves us the trouble of having to check - the user can handle the pooling themselves 😄

Reply:
Ah I see why you wanted this now, but I think for this level of the API, it is friendlier to "just accept the thing."